kitsune/sumo/static/sumo/js/tests/form-wizard-sign-in-step-tests.js

```diff
@@ -0,0 +1,305 @@
+    if (element.type == "submit") {
+      continue;
+    }
+
+    expect(expectedElements[element.name], `Checking ${element.name}`).to.exist;
+    expect(element.type).to.equal(expectedElements[element.name].type);
+    expect(element.disabled).to.equal(expectedElements[element.name].disabled);
+    expect(element.value).to.equal(expectedElements[element.name].value);
+  }
+}
+
+describe("sign-in-step custom element", () => {
+  let step;
+  let slot;
+
+  beforeEach(() => {
+    step = new SignInStep();
+    $("body")
+      .empty()
+      .append(step);
+  });
+
+  it("should render a sign-in-step custom element", () => {
+    expect(step).to.exist;
+    expect(step).to.be.an.instanceof(SignInStep);
+  });
+
+  it("should disable flow_id and flow_begin_time form fields if not yet supplied", () => {
+    const TEST_STATE_NO_FLOW_METRICS = {
+      utm_source: "utm_source",
+      utm_campaign: "utm_campaign",
+      utm_medium: "utm_medium",
+      entrypoint: "entrypoint",
+      entrypoint_experiment: "entrypoint_experiment",
+      entrypoint_variation: "entrypoint_variation",
+      context: "context",
+      redirect_to: window.location.href,
+      redirect_immediately: true,
+    };
+
+    const EXPECTED_FORM_ELEMENTS_NO_FLOW_METRICS = {
+      utm_source: { type: "hidden", value: "utm_source", disabled: false },
+      utm_campaign: { type: "hidden", value: "utm_campaign", disabled: false },
+      utm_medium: { type: "hidden", value: "utm_medium", disabled: false },
+      entrypoint: { type: "hidden", value: "entrypoint", disabled: false },
+      entrypoint_experiment: { type: "hidden", value: "entrypoint_experiment", disabled: false },
+      entrypoint_variation: { type: "hidden", value: "entrypoint_variation", disabled: false },
+      context: { type: "hidden", value: "context", disabled: false },
+      redirect_to: { type: "hidden", value: window.location.href, disabled: false },
+      redirect_immediately: { type: "hidden", value: "true", disabled: false },
+
+      service: { type: "hidden", value: "sync", disabled: false },
+      action: { type: "hidden", value: "email", disabled: false },
+      flow_id: { type: "hidden", value: "", disabled: true },
+      flow_begin_time: { type: "hidden", value: "", disabled: true },
+      email: { type: "email", value: "", disabled: false },
+    };
+
+    step.setState(TEST_STATE_NO_FLOW_METRICS);
+
+    let form = step.shadowRoot.querySelector("form");
+    assertFormElements(form, EXPECTED_FORM_ELEMENTS_NO_FLOW_METRICS);
+
+    // Now add the flow metrics, which should enable the fields.
+
+    const TEST_STATE_WITH_FLOW_METRICS = {
+      utm_source: "utm_source",
+      utm_campaign: "utm_campaign",
+      utm_medium: "utm_medium",
+      entrypoint: "entrypoint",
+      entrypoint_experiment: "entrypoint_experiment",
+      entrypoint_variation: "entrypoint_variation",
+      context: "context",
+      redirect_to: window.location.href,
+      redirect_immediately: true,
+
```

Comment:
```suggestion
      ...TEST_STATE_NO_FLOW_METRICS,
```

Totally optional, it can also be nice to have the data fully laid out in tests 🤷🏻